File: java/src/main/java/com/dddeurope/niuz/website/TeaserDto.java
package com.dddeurope.niuz.website;

import java.util.Objects;

public class TeaserDto {
    private final String headline;
    private final String author;

    public TeaserDto(String headline, String author) {
        this.headline = headline;
        this.author = author;
    }

    @Override
    public String toString() {
        return "TeaserDto{" +
                "headline='" + headline + '\'' +
                ", author='" + author + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TeaserDto teaserDto = (TeaserDto) o;
        return Objects.equals(headline, teaserDto.headline) && Objects.equals(author, teaserDto.author);
    }

    @Override
    public int hashCode() {
        return Objects.hash(headline, author);
    }
}

File: dotnet/niuz.application/website/TeaserDto.cs
using System;

namespace niuz.application.website
{
    public class TeaserDto
    {
        private readonly string headline;
        private readonly string author;

        public TeaserDto(string headline, string author)
        {
            this.headline = headline;
            this.author = author;
        }

        protected bool Equals(TeaserDto other)
        {
            return headline == other.headline && author == other.author;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((TeaserDto) obj);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(headline, author);
        }
    }
}